internal/server/kong/ws/config/version_compatibility_test.go

```diff
@@ -6274,6 +6274,95 @@ func TestVersionCompatibility_ProcessConfigTableUpdates(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "removing whole top-level fields",
```

Comment:
Can you please add a couple of more tests:
- test that `KeySet` is removed when `< 3.1.0`
- test that `KeySet` is not removed when `>= 3.1.0`
- test that `Key` is removed when `>= 3.1.0`

Reply:
I added that test for the `TopLevel` operation type I wrote; but it was noted that that was redundant with the `Remove: true` flag, so this test is actually redundant with the "remove core entities when indicated", and the others you mention are equivalent to the "don't remove core entities when not indicated"